### DISCUSSION TITLE
Save a drawing response in an image format

### DISCUSSION THREAD

**Discussion by joseluisulloafulgeri** [jspsych/jsPsych#2749] [Q&A]
hello,

For an experiment we want to ask people to draw a line over an image,
We managed to do this in Javascript, and we can recover the trace if the participant download the image (with the trace) and send it to us,

We were wondering if there is any way we can automatically save the image once the participant has responded (for instance in Cognition.run),

Any hints will be appreciated,
best,
José

**Comment by becky-gilbert** [comment]
Hi @joseluisulloafulgeri, are you using the jsPsych sketchpad plugin for this drawing trial? If so, and if `save_final_image` is `true` (the default), then the image will be saved to the plugin's [trial data](https://www.jspsych.org/7.3/plugins/sketchpad/#data-generated) as a base64-encoded string. Then you can just use a base64-to-image decoder to turn the string representation back into an image. There are online converters you can use, or you can do the conversion in a language like R or Python (see links toward the top of the [sketchpad plugin docs page](https://www.jspsych.org/7.3/plugins/sketchpad/)). This plugin also has a `save_strokes` parameter, which will save all of the individual stroke data, in case that's useful.

Or if you're using a custom plugin, then you can try using the [`toDataURL` method](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toDataURL) to convert the canvas image into a base64 string. Here's the relevant line in the sketchpad plugin, where `sketchpad` is the canvas element:
https://github.com/jspsych/jsPsych/blob/e5447cee27734d0352b4f8f15ec1ac99b3f6927a/packages/plugin-sketchpad/src/index.ts#L659
Does that help?

**Reply by joseluisulloafulgeri** [reply]
Thanks a lot Becky! We are actually using the canvas function, but we will explore your suggestion,
